jupyter_console/interactiveshell.py

```diff
@@ -109,12 +109,23 @@ class ZMQTerminalInteractiveShell(TerminalInteractiveShell):
         """
     )
 
+    use_kernel_is_complete = Bool(True, config=True,
+        help="""Whether to use the kernel's is_complete message
+        handling. If False, then the frontend will use its
+        own is_complete handler.
+        """
+    )
```

Comment:
I don't think we should have a config option for this: the old way of doing it in the frontend assumes that the code being entered is in Python, and most cases where the kernel doesn't support `is_complete` are non Python. So it's the wrong choice to offer the user.

For now, it's probably sensible to have some way to disable the `is_complete` requests, because the delay for kernels that don't implement it might be annoying. But I would do something much simpler than `input_splitter` - either always execute when a line is entered, or prompt for another line until the last line is blank (i.e. you press enter twice to execute).

Eventually it would be good for the frontend to be able to tell when the kernel doesn't support `is_complete` and activate such fallback behaviour automatically, but that needn't be done now.